### PR TITLE
[BugFix] Fix SyncDataCollector reset

### LINKED ISSUE #570: [BUG] SyncDataCollector resets all workers instead of one
## Describe the bug

When using the SyncDataCollector with a ParallelEnv, **all** workers are reset if any of the workers is done.

## To Reproduce

```python
import torch
from torchrl.collectors import SyncDataCollector
from torchrl.envs import ParallelEnv
from torchrl.envs.libs.gym import GymEnv

if __name__ == '__main__':
    num_workers = 2
    num_steps = 4000

    create_env = lambda: GymEnv(env_name='PongNoFrameskip-v4', frame_skip=4)
    create_parallel_env = lambda: ParallelEnv(num_workers, create_env)

    collector = SyncDataCollector(create_parallel_env, frames_per_batch=num_workers, split_trajs=False)
    iterator = collector.iterator()
    episode_rewards = []
    cumulative_reward = torch.zeros(num_workers)
    for _ in range(num_steps):
        batch = next(iterator).squeeze(1)
        cumulative_reward += batch['reward'].squeeze(-1)
        done = batch['done'].squeeze(-1)
        episode_rewards.extend(cumulative_reward[done].tolist())
        cumulative_reward[done] = 0.
    collector.shutdown()
    print(episode_rewards)
```

This leads to the following episode rewards, although the lowest value for Pong should be -21:
```bash
[-21.0, -37.0, -41.0, -21.0]
```
The reason is that some episodes are reset without notice.

## Expected behavior

Only the workers which are actually done should be reset.

## Reason and Possible fixes

[In this line](https://github.com/pytorch/rl/blob/4862459866153a499a3c7e5d6df4238ffdc29fe8/torchrl/collectors/collectors.py#L588) the ParallelEnv is reset but `"reset_workers"` is ignored.  
A quick fix would be to replace the line with:
```python
self._tensordict.update(self.env.reset(self._tensordict), inplace=True)
```
This leads to the following output for the above code and seems to be correct:
```bash
[-21.0, -21.0, -21.0, -21.0, -21.0, -20.0, -20.0, -21.0]
```
But maybe there is a better solution.

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have read the [documentation](https://github.com/pytorch/rl/tree/main/docs/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

### PR DESCRIPTION
**Pull request by jrobine**
## Description

Fix #570 
`close #570` 

- [x] I have raised an issue to propose this change

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: torchrl/collectors/collectors.py [hunk 1]
```diff
@@ -585,7 +585,7 @@ def _reset_if_necessary(self) -> None:
             else:
                 self._tensordict.zero_()
 
-            self._tensordict.update(self.env.reset(), inplace=True)
+            self._tensordict.update(self.env.reset(self._tensordict), inplace=True)
```
Comment (vmoens) [inline torchrl/collectors/collectors.py:588]: I think we don't need the call to update.
env.reset() will update the tensordict.

**Review by vmoens — COMMENTED** [review]
Thanks! Let's try without the update?

**Comment by vmoens** [comment]
I implemented some tests, let me know if they make sense

**Comment by jrobine** [comment]
Thanks! It seems like the step counts are also incorrect, so I think only assertions 313 and 322 are really necessary.

**Comment by vmoens** [comment]
Do you mean "were incorrect" or are they still incorrect after this PR?

**Comment by jrobine** [comment]
Sorry, they _were_ incorrect

**Pull request merged**